### PR TITLE
Document how to control parallelism. Fixes #20.

### LINKED ISSUE #20: Document how to control parallelism
How is the number of reducers set?

### PR DESCRIPTION
**Pull request by tomwhite**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +142,20 @@ hadoop jar target/quince-0.0.1-SNAPSHOT-job.jar \
...
 
+### Controlling parallelism
+
+The `LoadVariantsTool` runs using a single MapReduce shuffle. The number of map tasks is 
+determined by the size of the input, with one map being created for each HDFS block 
+(usually 128MB) of input. Normally there's no reason to change this, but it can be 
+adjusted by setting the Hadoop property `mapred.max.split.size` to a smaller value (in 
+bytes).
+
+The number of reduce tasks is determined by automatically (by Crunch) so that each
```
Comment (laserson) [inline README.md]: delete extra "by"

File: src/main/java/com/cloudera/science/quince/FlattenVariantFn.java [hunk 2]
```diff
@@ -84,6 +84,15 @@ public void process(Variant variant, Emitter<FlatVariantCall> emitter) {
       }
     }
   }
+
+  @Override
+  public float scaleFactor() {
+    // Factor of two is an underestimate since it assumes that variants and calls are
+    // about the same size (in bytes). It also doesn't take into account the number of
+    // samples (calls) written compared to the number in the original variant.
+    return variantsOnly ? super.scaleFactor() : 2.0f;
```
Comment (laserson) [inline src/main/java/com/cloudera/science/quince/FlattenVariantFn.java]: Shouldn't this be a function of the number of samples in the case that we also have genotype calls?

File: README.md [hunk 3]
```diff
@@ -142,6 +142,20 @@ hadoop jar target/quince-0.0.1-SNAPSHOT-job.jar \
...
+### Controlling parallelism
+
+The `LoadVariantsTool` runs using a single MapReduce shuffle. The number of map tasks is 
+determined by the size of the input, with one map being created for each HDFS block 
+(usually 128MB) of input. Normally there's no reason to change this, but it can be 
+adjusted by setting the Hadoop property `mapred.max.split.size` to a smaller value (in 
+bytes).
+
+The number of reduce tasks is determined by automatically (by Crunch) so that each 
+reduce task has approximately 1GB of input. There is a maximum of 500 reducers
```
Comment (laserson) [inline README.md]: Where does the 1GB of input come in?  Is this a Crunch thing?

File: README.md (hunk 1 above)
Comment (tomwhite) [inline README.md]: Done

File: README.md (hunk 3 above)
Comment (tomwhite) [inline README.md]: Yes. I added a mention of the property that controls it.

File: src/main/java/com/cloudera/science/quince/FlattenVariantFn.java (hunk 2 above)
Comment (tomwhite) [inline src/main/java/com/cloudera/science/quince/FlattenVariantFn.java]: My reasoning was as follows. If the variant information takes _v_ bytes and call information _c_ bytes, then the size goes from _v + nc_ to _n(v + c)_ for _n_ calls. I assumed that _v_ and _c_ are about the same, giving a scale factor of 2. 

I just did a small experiment to measure the size, and _v_ is approximately twice _c_. This gives a factor of 3, which seems to work out a bit better. We could make the formula more complex, but this is a decent start, I think.

**Pull request merged**